src/main/java/com/facebook/openwifirrm/modules/ApiServer.java

```diff
@@ -269,16 +290,23 @@ private String getFullUrl(String path, String queryString) {
 	 * HTTP 403 response and return false.
 	 */
 	private boolean performOpenWifiAuth(Request request, Response response) {
-		// TODO check if request came from internal endpoint
-		boolean internal = true;
-		String internalName = request.headers("X-INTERNAL-NAME");
-		if (internal && internalName != null) {
-			// Internal request, validate "X-API-KEY"
-			String apiKey = request.headers("X-API-KEY");
-			if (apiKey != null && apiKey.equals(serviceKey)) {
-				// auth success
-				return true;
+		boolean internal = request.port() == params.internalHttpPort;
+		if (internal) {
+			// TODO if this doesn't exist, is it an error???
+			String internalName = request.headers("X-INTERNAL-NAME");
+			if (internalName != null) {
+				// Internal request, validate "X-API-KEY"
+				String apiKey = request.headers("X-API-KEY");
+				if (apiKey != null && apiKey.equals(serviceKey)) {
+					// auth success
+					return true;
+				}
 			}
+			// auth failure
+			// TODO normally 401s are expected to have WWW-Authenticate header but this
```

Comment:
this is still an open question for me

Reply:
owsec always throws 403 for me, might as well just do what they do. Let's avoid 401 and revisit if this changes upstream.

```
$ curl -i -k -H "Accept: application/json" -H "X-INTERNAL-NAME: asdf" -H "X-API-KEY: asdf" "h
ttps://owsec.wlan.local:17001/api/v1/system?command=info"
HTTP/1.1 403 OK
Date: ...
Connection: Keep-Alive
Transfer-Encoding: chunked
Content-Type: application/json
Access-Control-Allow-Origin: *
Vary: Origin, Accept-Encoding
Keep-Alive: timeout=30, max=1000

{"ErrorCode":8,"ErrorDescription":"8: Invalid token.","ErrorDetails":"GET"}
```